src/refdb_fs.c

```diff
@@ -466,10 +452,176 @@ static int ref_error_notfound(const char *name)
 	return GIT_ENOTFOUND;
 }
 
-static int packed_lookup(
-	git_reference **out,
-	refdb_fs_backend *backend,
-	const char *ref_name)
+static const char *packed_set_peeling_mode(
+        const char *data,
+        size_t data_sz,
+        refdb_fs_backend *backend)
+{
+	static const char *traits_header = "# pack-refs with:";
+	const char *eol;
+	backend->peeling_mode = PEELING_NONE;
+
+	if (data_sz == 0 || *data != '#') {
+		return data;
+	}
```

Comment:
```suggestion
	if (data_sz == 0 || *data != '#')
		return data;
```

Reply:
Actually ,looking at this again, this feels a little unnecessary?  `git__prefixncmp` should catch both of these cases, and quickly enough that a separate test shouldn't be necessary.  🤔

Reply:
I agree, I think we can remove this test entirely. Thanks!